plugins/base/src/main/kotlin/resolvers/anchors/AnchorsHint.kt

```diff
@@ -1,9 +1,13 @@
 package org.jetbrains.dokka.base.resolvers.anchors
 
+import org.jetbrains.dokka.model.Documentable
 import org.jetbrains.dokka.model.properties.ExtraProperty
 import org.jetbrains.dokka.pages.ContentNode
 
-// TODO IMPORTANT: https://github.com/Kotlin/dokka/issues/1054
-object SymbolAnchorHint: ExtraProperty<ContentNode>, ExtraProperty.Key<ContentNode, SymbolAnchorHint> {
-    override val key = this
+data class SymbolAnchorHint(val anchorName: String): ExtraProperty<ContentNode> {
+    object SymbolAnchorHintKey : ExtraProperty.Key<ContentNode, SymbolAnchorHint>
+    override val key: ExtraProperty.Key<ContentNode, SymbolAnchorHint> = SymbolAnchorHintKey
+    companion object: ExtraProperty.Key<ContentNode, SymbolAnchorHint> {
```

Comment:
What is the reason of this object implementing `ExtraProperty.Key` if it is not used as a key?

Reply:
I don't understand. It is since when i'd like to get values from extras i use it

Reply:
Fixed in the second PR